tests/backup/backup_helper.go

```diff
@@ -381,6 +388,51 @@ func FilterAppContextsByNamespace(appContexts []*scheduler.Context, namespaces [
 	return
 }
 
+func InsertDataForBackupValidation(namespaces []string, ctx context.Context, existingAppHandler []appDriver.ApplicationDriver, backupName string,
+	commandBeforeBackup map[appDriver.ApplicationDriver]map[string][]string) ([]appDriver.ApplicationDriver, map[appDriver.ApplicationDriver]map[string][]string, error) {
+
+	// afterBackup - Check if the data is being inserted before or after backup
+
+	// Getting app handlers for deployed apps in the namespace and inserting data to same
+	var err error
+	var allHandlers []appDriver.ApplicationDriver
+	var dataCommands = make(map[appDriver.ApplicationDriver]map[string][]string)
+
+	if len(existingAppHandler) == 0 {
+		for _, eachNamespace := range namespaces {
+			if handler, ok := NamespaceAppWithDataMap[eachNamespace]; ok {
+				log.InfoD("App with data support found under namespace - [%s]", eachNamespace)
+				for _, eachHandler := range handler {
+					log.InfoD("Adding data to app in namespace  - [%s]", eachNamespace)
+					dataCommands[eachHandler] = eachHandler.GetRandomDataCommands(queryCountForValidation)
+					err = eachHandler.InsertBackupData(ctx, backupName, dataCommands[eachHandler]["insert"])
```

Comment:
It is better to run in goroutines?

Reply:
Will file another task and take this up as part of it, for now it is not taking more than 5 sec to verify all the restores.